File: src/main/java/se/gigurra/gat/util/VertexArrayObject.java
package se.gigurra.gat.util;

import javax.media.opengl.GL2ES3;

public class VertexArrayObject {

	public VertexArrayObject(final GL2ES3 gl, final boolean enable) {
		int[] arr = new int[1];
		m_gl = gl;
		m_gl.glGenVertexArrays(1, arr, 0);
		m_object = arr[0];

		if (enable) {
			enable();
		}

	}

	public void enable() {
		m_gl.glBindVertexArray(m_object);
	}

	public void disable() {
		m_gl.glBindVertexArray(0);
	}

	public void dispose() {
		int[] arr = { m_object };
		m_gl.glDeleteVertexArrays(1, arr, 0);
	}

	public int object() {
		return m_object;
	}

	private int m_object;
	private GL2ES3 m_gl;
}

File: src/main/scala/se/gigurra/gat/util/VertexArrayObject.scala
package se.gigurra.gat.util

import Edit.EditCls
import javax.media.opengl.GL2ES3

class VertexArrayObject(gl: GL2ES3, enableNow: Boolean) {

  val glId = Array(0).edit(gl.glGenVertexArrays(1, _, 0))(0)

  if (enableNow) enable()

  def enable() { gl.glBindVertexArray(glId) }
  def disable() { gl.glBindVertexArray(0) }
  def dispose() { gl.glDeleteVertexArrays(1, Array(glId), 0) }

}
